### PR TITLE
XRDDEV-2145 API to get the certificate of a certification service

### PR DESCRIPTION
**Pull request by justasnortal**
API to get the certificate of a certification service.

https://nordic-institute.atlassian.net/browse/XRDDEV-2145

### CODE REVIEW

File: src/centralserver/admin-service/infra-api-rest/src/main/java/org/niis/xroad/centralserver/restapi/converter/CertificateDetailsDtoConverter.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * THE SOFTWARE.
+ */
+
+package org.niis.xroad.centralserver.restapi.converter;
+
+import org.mapstruct.Mapper;
+import org.niis.xroad.centralserver.openapi.model.CertificateDetailsDto;
+import org.niis.xroad.centralserver.restapi.dto.CertificateDetails;
+
+@Mapper(componentModel = "spring")
```
Comment (ricardas-buc) [inline src/centralserver/admin-service/infra-api-rest/src/main/java/org/niis/xroad/centralserver/restapi/converter/CertificateDetailsDtoConverter.java:34]: we should consider upgrading to mapstruct 1.5.x, then we could do
@Mapper(componentModel = MappingConstants.ComponentModel.SPRING)

File: src/centralserver/admin-service/core/src/main/java/org/niis/xroad/centralserver/restapi/dto/converter/CaInfoConverter.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import static java.lang.String.valueOf;
+
+@Component
+@RequiredArgsConstructor
+public class CaInfoConverter {
+
+    private final KeyUsageConverter keyUsageConverter;
+
+    @SneakyThrows
+    public CertificateDetails convert(CaInfo caInfo) {
```
Comment (ricardas-buc) [inline src/centralserver/admin-service/core/src/main/java/org/niis/xroad/centralserver/restapi/dto/converter/CaInfoConverter.java:55]: was use of mapstruct considered here?  I do see pros and cons of using it so curious regarding this decision

**Review by ricardas-buc — APPROVED** [review]

File: src/centralserver/admin-service/core/src/main/java/org/niis/xroad/centralserver/restapi/dto/converter/CaInfoConverter.java (hunk 2 above)
Comment (justasnortal) [reply src/centralserver/admin-service/core/src/main/java/org/niis/xroad/centralserver/restapi/dto/converter/CaInfoConverter.java:55]: Yes, it was. But when mapping from entity to dto (dto fields naming is based on openapi dto), many fields would be mapped using annotations. The amount of code lines would be similar, but it is more error prone.

File: src/centralserver/admin-service/infra-api-rest/src/main/java/org/niis/xroad/centralserver/restapi/converter/CertificateDetailsDtoConverter.java (hunk 1 above)
Comment (justasnortal) [reply src/centralserver/admin-service/infra-api-rest/src/main/java/org/niis/xroad/centralserver/restapi/converter/CertificateDetailsDtoConverter.java]: Updated

**Comment by justasnortal** [comment]
[rerun-ci]

**Pull request merged**